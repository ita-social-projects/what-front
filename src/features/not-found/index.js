export { NotFound } from './not-found.js';