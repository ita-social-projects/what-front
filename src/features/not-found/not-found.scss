.wrapper {
  position: absolute;
  width: 100%;
  top: 0;
  overflow: hidden;
  min-height: 100%;
  background-color: #01073A;
}

.error-container {
  background-color: #fff;
  margin-top: 30vh;
  border-radius: 3px;
  max-width: 500px !important;
  transition: all ease .5s;

  @media (max-width: 768px) {
    margin-top: 20vh;
    transition: all ease .5s;
  }

  @media (max-width: 576px) {
    margin-top: 10vh;
    transition: all ease .5s;
  }
}