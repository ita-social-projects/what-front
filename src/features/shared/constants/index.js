export * from './proptypes-shapes.js';
