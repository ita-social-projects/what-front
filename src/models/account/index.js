export { accountReducer } from './reducer.js';
export {
  authWatcher, login, registration, logOut, fetchUsersList, fetchUnAssignedUserList,
  clearRegistration, newPassword,
} from './sagas.js';
export {
  currentUserSelector, newUserSelector, fetchAssignedUsersSelector, registrationSelector,
  changePasswordSelector,
} from './selectors.js';
