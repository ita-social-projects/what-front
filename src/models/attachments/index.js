export { attachmentsReducer } from './reducer.js';
export * from './actions.js';
export {
  attachmentsSelector,
  attachmentByIdSelector,
  attachmentsCreatingSelector,
} from './selectors.js';