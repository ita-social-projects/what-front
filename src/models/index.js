export * from './account/index.js';
export * from './mentors/index.js';
export * from './courses/index.js';
export * from './themes/index.js';
export * from './schedules/index.js';
export * from './students/index.js';
export * from './lessons/index.js';
export * from './secretary/index.js';
export * from './student-groups/index.js';