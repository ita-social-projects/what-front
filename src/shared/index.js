export * from './hooks';
export * from './api-service';
export * from './routes';
