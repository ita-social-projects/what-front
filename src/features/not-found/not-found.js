import React from 'react';
import className from 'classnames';
import { useHistory } from 'react-router-dom';
import { useSelector, shallowEqual } from 'react-redux';
import { homepages } from '@/shared';
import { currentUserSelector } from '@/models';
import styles from './not-found.scss';

import { Button } from '../../components/index.js';

export const NotFound = () => {
  const history = useHistory();
  const { currentUser } = useSelector(currentUserSelector, shallowEqual);

  return (
    <div className="container">
      <div className={className(styles['error'],'text-center')}>
        <div className="col-7">
          <h1 className={className(styles['title'])}>404</h1>
          <p className={className(styles['text'])}> This page is missing</p>
          <Button type="button" onClick={() => (history.push(homepages[currentUser.role]))}
                   className={className(styles['home-button'], 'mt-4')}>HOME</Button>
        </div>
      </div>
    </div>);
};
