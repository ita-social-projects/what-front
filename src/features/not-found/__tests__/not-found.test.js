import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { NotFound } from '../not-found.js';
import { currentUserSelector } from '@/models';
import { homepages } from '@/shared';

jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue({
    currentUser: jest.fn().mockReturnValue({
      role: 4,
    }),
  }),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn().mockReturnValue({ push: jest.fn() }),
}));

describe('NotFound', () => {
  it('renders NotFound component', () => {
    const { container } = render(
      <Router>
        <NotFound />
      </Router>
    );
    const notFoundContainer = container.getElementsByClassName('container');

    expect(notFoundContainer).toMatchSnapshot();
  });

  it('user navigates to page by role', () => {
    const history = useHistory();
    const { currentUser } = useSelector(currentUserSelector);

    render(
      <Router>
        <NotFound />
      </Router>
    );

    history.push(homepages[currentUser.role]);
    expect(history.push).toHaveBeenCalledWith(homepages[currentUser.role]);
  });
});
