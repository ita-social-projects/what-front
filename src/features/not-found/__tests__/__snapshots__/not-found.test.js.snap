// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders NotFound component 1`] = `
HTMLCollection [
  <div
    class="container"
  >
    <div
      class="text-center"
    >
      <div
        class="col-7"
      >
        <h1
          class=""
        >
          404
        </h1>
        <p
          class=""
        >
           This page is missing
        </p>
        <button
          class="btn undefined mt-4"
          type="button"
        >
          HOME
        </button>
      </div>
    </div>
  </div>,
]
`;
